hawk/src/main/java/com/orhanobut/hawk/Hawk.java

```diff
@@ -178,4 +239,97 @@ public static boolean resetCrypto() {
     public static LogLevel getLogLevel() {
         return logLevel;
     }
+
+    /**
+     * Provides the ability to chain put invocations, for example
+     * <code>Hawk.chain().put("foo", 0).put("bar", 1).done()</code>
+     */
+    public static final class HawkChain {
+
+        private final List<Pair<String, ?>> items;
+        private boolean atomic;
+
+        public HawkChain() {
+            this(4);
+        }
+
+        public HawkChain(int capacity) {
+            items = new ArrayList<>(capacity);
+        }
+
+        /**
+         * Saves every type of Objects. List, List<T>, primitives
+         *
+         * @param key   is used to save the data
+         * @param value is the data that is gonna be saved. Value can be object, list type, primitives
+         */
+        public <T> HawkChain put(String key, T value) {
+            String fullText = Hawk.put(key, value, null, false, false);
+            addItem(key, fullText);
+            return this;
+        }
+
+        /**
+         * Saves the list of objects to the storage
+         *
+         * @param key  is used to save the data
+         * @param list is the data that will be saved
+         */
+        public <T> HawkChain put(String key, List<T> list) {
+            String fullText = Hawk.put(key, null, list, true, false);
+            addItem(key, fullText);
+            return this;
+        }
+
+        /**
+         * If true, Hawk throws an exception if any of the chained invocations fails.
+         * Disabled by default.
+         *
+         * @param atomic true to throw an exception, false otherwise.
+         */
+        public HawkChain atomic(boolean atomic) {
+            this.atomic = atomic;
+            return this;
+        }
+
+        /**
+         * Saves the chained values.
+         */
+        public boolean done() {
+            return storage.put(items);
+        }
+
+        /**
+         * Saves the chained values and returns the saved keys.
+         *
+         * @return the set of saved keys.
+         */
+        public Set<String> doneWithKeys() {
+            boolean done = storage.put(items);
+
+            Set<String> keys;
+            if (done) {
+                keys = new HashSet<>(items.size());
+                for (Pair<String, ?> p : items) {
+                    keys.add(p.first);
+                }
+            } else {
+                keys = Collections.emptySet();
+            }
+
+            return keys;
+        }
+
+        private void addItem(String key, String data) {
```

Comment:
data should be already validated before addItem. This method should do only 
<pre>
items.add(new Pair<>(key, data));
</pre>